evalml/pipelines/components/transformers/preprocessing/datetime_featurizer.py

```diff
@@ -102,7 +103,10 @@ def transform(self, X, y=None):
                 X_t[name] = features
                 if categories:
                     self._categories[name] = categories
-        X_t = X_t.drop(self._date_time_col_names, axis=1)
+        col_names_to_drop = self._date_time_col_names.copy()
```

Comment:
if we don't drop the date index will pipelines built by AutoMLSearch work? Right now we add a `datetime_featurizer` if there is a datetime column:

```datetime_cols = X.select(["Datetime"])
    add_datetime_featurizer = len(datetime_cols.columns) > 0
    if add_datetime_featurizer:
        pp_components.append(DateTimeFeaturizer)
```

If it works I think it would be good to see a test case on it!

Reply:
I see below we have tests where `fit` is mocked out so it might be good just to make sure pipelines fit as well. My concern is that other components will not be able to handle the datetime column and will fail there.

Reply:
Agreed, so the plan is to go back to the previous implementation of `DateTimeFeaturizer` and drop all datetime columns, and don't add `DateTimeFeaturizer` or `DelayedFeatureTransformer` to ARIMA during pipeline construction